sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4281,6 +4281,18 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
         Row(2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22) :: Nil)
     }
   }
+
+  test("SPARK-38182: Fix NoSuchElementException if pushed filter does not contain any " +
```

Comment:
And, if you don't mind, could you add a minimal test coverage at `FileIndexSuite.scala` instead of `SQLQuerySuite`, @ulysses-you ?

Reply:
thank you @dongjoon-hyun for the reminder !